src/code/z_actor.c

```diff
@@ -2258,27 +2275,44 @@ void func_8003115C(GlobalContext* globalCtx, s32 numInvisibleActors, Actor** inv
     gDPPipeSync(POLY_XLU_DISP++);
 
     if (globalCtx->roomCtx.curRoom.showInvisActors == 0) {
+        // setup to update the zbuffer (`Z_UPD`)
+        // and the framebuffer:
+        // `G_RM_CLD_SURF | G_RM_CLD_SURF2` sets the blender to blend what is being rendered into the framebuffer
         gDPSetOtherMode(POLY_XLU_DISP++,
                         G_AD_DISABLE | G_CD_MAGICSQ | G_CK_NONE | G_TC_FILT | G_TF_BILERP | G_TT_NONE | G_TL_TILE |
                             G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
                         G_AC_THRESHOLD | G_ZS_PRIM | Z_UPD | G_RM_CLD_SURF | G_RM_CLD_SURF2);
+        // makes the drawn image the color of the primcolor, red
         gDPSetCombineMode(POLY_XLU_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
         gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 0, 0, 255);
+        // the z-buffer will later only allow drawing inside the lens circle
     } else {
+        // setup to update the zbuffer (Z_UPD)
+        // but leave the framebuffer untouched:
+        // `G_BL_CLR_BL, G_BL_0, G_BL_CLR_MEM, G_BL_1MA` sets the blender to leave the framebuffer untouched
```

Comment:
Im still not very much a fan of the very verbose docs of what the gfx stuff is doing. feels like something better explained outside of the code. Or if we must have it in the code, something a bit more brief.

I understand that this is complex and that the detail is probably desired. Which is why external documentation seems more appropriate in this case to me

Reply:
everything else in the pr seems fine to me though

Reply:
I think it's good to keep until we get those more extensive docs in either the GBI file or another central place. Cause yeah the SDK docs are nice but there aren't really examples in them.

Reply:
maybe to some extent, but we can scale back the verbosity a bit. For example i dont think its necessary to comment on a prim color macro saying its red.

Reply:
https://github.com/louist103/oot/pull/8

Reply:
with the changes in the pr this would be ready to approve from me